examples/websocket/data_uploader/server/server.go

```diff
@@ -0,0 +1,103 @@
+package main
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"io"
+	"log"
+	"net/http"
+	"os"
+	"os/signal"
+	"time"
+
+	"github.com/lesismal/nbio/nbhttp"
+	"github.com/lesismal/nbio/nbhttp/websocket"
+)
+
+var compression = flag.Bool("compression", false, "allow compression")
+
+func newUpgrader() *websocket.Upgrader {
+	u := websocket.NewUpgrader()
+	if *compression {
+		u.EnableCompression(true)
+	}
+	var output io.WriteCloser
+	id := 0
+	u.OnDataFrame(func(c *websocket.Conn, messageType websocket.MessageType, fin bool, data []byte) {
+		if output == nil {
+			var err error
+			deflateStr := ""
+			if u.CompressionEnabled() {
```

Comment:
Here should use c.remoteCompressionEnabled, because the Upgrader.enableCompression is just a config, it is decided both by the server and the client during the handshake:
https://github.com/lesismal/nbio/blob/master/nbhttp/websocket/upgrader.go#L191
https://github.com/lesismal/nbio/blob/master/nbhttp/websocket/upgrader.go#L265